Expandable Block composition

LostABike opened the pull request:
<!--
Thank you for your contribution to iTwinUI.

If you are only making changes to the docs site using the "Edit page on GitHub" link,
then you can ignore most of this template.
-->

## Changes

<!--
What kind of code changes does this PR include?
Mention anything that could be helpful for reviewers and include screenshots for visual changes.
-->

- Converted `ExpandableBlock` component to be a composition of subcomponents as follows:

```
   <ExpandableBlock>
     <ExpandableBlock.Header>
       <ExpandableBlock.LabelArea>
         <ExpandableBlock/>
         <ExpandableBlock.Caption/>
       </ExpandableBlock.LabelArea>
       <ExpandableBlock.EndIcon/>
     </ExpandableBlock.Header>
     <ExpandableBlock.Content/>
   </ExpandableBlock>
```

- ~~Changed `iui-header` class name to `iui-expandable-header`~~ reverted this cuz bug and to work on other stuff first, will come back.
- Updated css to apply @FlyersPh9 's solution for user's issue where expander icon changes for nested block when it shouldn't be. Verified working!
- Updated Storybook to use composition.
- HTML looks fine.

Closes #1292 

## Testing

<!--
How did you test your changes?
If your PR has visual changes, then make sure they are demonstrated in html test pages as well as
storybook, then approve visual test images for both (`yarn approve:css` and `yarn approve:react`).

If not applicable, you can write "N/A".
-->
- Manually tested all cases in Vite and Storybook, verified working as old component.
- Updated unit tests.

## Docs

<!--
If your PR includes user-facing changes, then update docs in all places (JSDoc, website, stories, etc).
Make sure to include a changeset (`yarn changeset`).

If not applicable, you can write "N/A".
-->
- Added jsdocs for all subcomponents
- TBD: ~~Website examples~~ + docs update

mayank99 commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -120,43 +179,146 @@ export const ExpandableBlock = React.forwardRef((props, ref) => {
+      ref={forwardedRef}
       {...rest}
     >
+      <SvgChevronRight className='iui-icon' aria-hidden />
```
if this icon is not customizable, then maybe it should be moved into the css. 🤔

LostABike replied:
Yea I was also thinking about making it customizable. Updated. But then it also made the code too extra since it is part of header (aka if header has children then all the label/title code must also be added). So I decided to make `expandIcon` a convenience prop of `Header`, and if it is passed in then we would just provide the default icon. Otherwise user can override with the `ExpandIcon` subcomponent to use customize the icon they want.

mayank99 commented on packages/itwinui-css/src/expandable-block/expandable-block.scss:
any chance we can change the css class names to be more specific? e.g. `iui-header` -> `iui-expandable-block-header`. `iui-icon` -> `iui-expandable-block-icon`, etc

mayank99 replied:
what about `iui-small` and `iui-borderless`? those could be data attributes

mayank99 replied:
bump

LostABike replied:
updated! changeset as well

mayank99 commented on apps/storybook/src/ExpandableBlock.stories.tsx:
```diff
@@ -29,29 +28,38 @@ export default {
-      {args.children}
+    <ExpandableBlock {...args}>
+      <ExpandableBlock.Header>
+        <ExpandableBlock.LabelArea>
```
this is a lot of code for a basic block. Maybe a convenience `label` prop in `ExpandableBlock.Header` will help (similar to `Tile.Name`)

```jsx
<ExpandableBlock>
	<ExpandableBlock.Header label='Blah' />
	<ExpandableBlock.Content>
	 ...
	</ExpandableBlock.Content>
</ExpandableBlock>
```

LostABike replied:
Done. I used the other subcomponent to keep the class structure the same.

gretanausedaite commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -120,43 +182,150 @@ export const ExpandableBlock = React.forwardRef((props, ref) => {
-          {caption && <Box className='iui-caption'>{caption}</Box>}
-        </Box>
-        {icon && <Icon fill={status}>{icon}</Icon>}
+        <Box>{children}</Box>
```
Why do you have this additional Box for children?

LostABike replied:
Because of this in the css file:
![image](https://github.com/iTwin/iTwinUI/assets/22462084/b43463ad-44b6-4f98-95d9-9450dd3ad577)

gretanausedaite commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -10,28 +10,58 @@ import {
-   * Small note displayed above title, regardless of state.
-   */
-  caption?: React.ReactNode;
+const ExpandableBlockContext = React.createContext<
```
You can reuse ExpandableBlockOwnProps to not repeat all of the already existing props.

LostABike replied:
How do you mean?

gretanausedaite replied:
const ExpandableBlockContext = ExpandableBlockOwnProps & {[props that are missing from main prop type]}

gretanausedaite reviewed (commented):
Can you do a11y testing?

LostABike commented:
> Can you do a11y testing?

Did you mean manually test NVDA on components with `aria-`?

gretanausedaite commented:
> Did you mean manually test NVDA on components with `aria-`?

NVDA screen reader and keyboard navigation

FlyersPh9 commented on packages/itwinui-css/src/expandable-block/expandable-block.scss:
This looks good to me, matches what is in https://github.com/iTwin/iTwinUI/pull/1376/files#diff-7139d3558365a97d23934152a27c2e4c8faba3aaeb01912529f15c90f764b7a2.

LostABike commented:
> > Did you mean manually test NVDA on components with `aria-`?
> 
> NVDA screen reader and keyboard navigation

Tested and verified everything working! One thing though, is that the status `Icon` doesn't have an `aria-label`, and I notice that only `IconButton` have those. Should we add label for `Icon` in this case? Since they convey a status but not necessarily a button.

![image](https://github.com/iTwin/iTwinUI/assets/22462084/4f5674ab-a846-4647-8236-c67fb589619e)

gretanausedaite commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -120,43 +183,174 @@ export const ExpandableBlock = React.forwardRef((props, ref) => {
+const ExpandableBlockTitle = React.forwardRef((props, forwardedRef) => {
+  const { className, children, ...rest } = props;
+  return (
+    <Box className={cx('iui-title', className)} ref={forwardedRef} {...rest}>
```
Maybe it's possible to change classNames to be unique? Eg. `iui-expandable-block-title`

LostABike replied:
Updated!

gretanausedaite commented on playgrounds/vite/src/App.tsx:
```diff
@@ -1,10 +1,17 @@
+import { ExpandableBlock } from '@itwin/itwinui-react';
 
 const App = () => {
+  // const onToggleMock = jest.fn();
```
You can remove playground code from PR

LostABike replied:
Oops, reverted :D

gretanausedaite commented:
> Should we add label for `Icon` in this case? Since they convey a status but not necessarily a button.

Icons are decorative elements that have aria-hidden option added.

LostABike commented:
Changing class names appear to cause visual tests to fail on PR but not locally?

LostABike commented:
I'm not sure why changing class names are causing the tests to mess up, seem to look fine locally in Storybook

LostABike commented:
Updated everything in response to all comments and more: 
- Changed class names to be more specific
- Tested accessibility
- Updated unit tests and html accordingly
- Shortened props declaration in context
- Make expandable icon into subcomponent
- Updated Website examples to use subcomponents (manually tested and verified working as before)

Side note: I believe the actual documentation text on website does not need to be updated since they describe usage more than implementations of the component?

To be updated:
- Migration guide
- Changesets: currently having trouble with this due to the following error:
![image](https://github.com/iTwin/iTwinUI/assets/22462084/2c7befeb-437a-44bb-a5d2-8209a61e80aa)

gretanausedaite reviewed (approved):
LGTM

mayank99 commented:
> > > Did you mean manually test NVDA on components with `aria-`?
> > 
> > 
> > NVDA screen reader and keyboard navigation
> 
> Tested and verified everything working!

Can you post a screen recording testing all states (expanded, collapsed, disabled)? See related thread: https://github.com/iTwin/iTwinUI/pull/1321#discussion_r1213522621

---

>     * Changesets: currently having trouble with this due to the following error:
>       ![image](https://user-images.githubusercontent.com/22462084/248515639-2c7befeb-437a-44bb-a5d2-8209a61e80aa.png)

Are you still facing this issue? Should have been fixed after `main` was merged into `dev` on Monday.

mayank99 commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -120,43 +154,178 @@ export const ExpandableBlock = React.forwardRef((props, ref) => {
-        },
-        className,
+      className={cx('iui-expandable-header', className)}
+      role='button'
```
this might be a good time to use a button instead of a div. See https://github.com/iTwin/iTwinUI/pull/1321#discussion_r1213349992

LostABike replied:
I updated this to a button and removed the `role='button'`, assuming it's not needed anymore

mayank99 replied:
yes, this is good. Some additional notes:
1. it should be possible for it to be both expanded and disabled at the same time.
2. tabindex is not needed because `<button>`s are already focusable.
3. onKeyDown is not needed because it is already handled when using `<button>` with `onClick`.
4. user's `onClick` prop needs to be combined with your handler.
5. add `type='button'` to prevent accidental form submissions

```jsx
  const { className, children, label, onClick: onClickProp, ...rest } = props;
  const { isExpanded, setExpanded, disabled, onToggle } = useSafeContext(
    ExpandableBlockContext,
  );

  return (
    <Box
      as='button'
      type='button'
      className={cx('iui-expandable-header', className)}
      aria-expanded={isExpanded}
      aria-disabled={disabled}
      onClick={mergeEventHandlers(onClickProp, () => {
        if (disabled) {
          return;
        }
        setExpanded(!isExpanded);
        onToggle?.(!isExpanded);
      })}
      ref={forwardedRef}
      {...rest}
    >
      {children ?? (
        <>
          <ExpandableBlock.ExpandIcon />
          <ExpandableBlock.LabelArea>
            <ExpandableBlock.Title>{label}</ExpandableBlock.Title>
          </ExpandableBlock.LabelArea>
        </>
      )}
    </Box>
  );
```

after all this, test again in NVDA to confirm behavior is unchanged.

LostABike replied:
Thanks for the suggestion. I added these changes and re-tested behavior + NVDA, working as expected still

mayank99 replied:
tabindex needs to be removed

LostABike replied:
removed

LostABike replied:
I am having trouble passing the `should trigger onToggle when clicked with Enter or Spacebar` unit test now because we eliminated the keyDown handler. I tried to add it back in this code but `mergeEventHandlers` would only accept MouseEvent argument and not KeyBoard, what do? It's worth noting that I have manually tested keyboard events (space, enter) on the header with a custom `onToggle` function and can verify that `onToggle` does get triggered, so I'm not sure why this test doesn't pass :-s

mayank99 commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -120,43 +154,178 @@ export const ExpandableBlock = React.forwardRef((props, ref) => {
-          {caption && <Box className='iui-caption'>{caption}</Box>}
-        </Box>
-        {icon && <Icon fill={status}>{icon}</Icon>}
+        <Box>{children}</Box>
```
lets add `innerProps` if user wants to add a class to this div

LostABike replied:
where is `innerProps` passed in from? Do you mean like this? 
```
const ExpandableBlockContent = React.forwardRef((props, forwardedRef) => {
  const { className, children, innerProps, ...rest } = props;
  ```

mayank99 replied:
yes, and used here:

```jsx
<Box {...innerProps}>{children}</Box>
```

passed by user like this:

```jsx
<ExpandableBlock.Content innerProps={{ className: 'my-inner' }}>
```

LostABike replied:
Ok I added innerProps

mayank99 replied:
This will be better:
```
innerProps: React.ComponentPropsWithoutRef<'div'>;
```

LostABike replied:
thanks! updated

mayank99 commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -120,43 +154,178 @@ export const ExpandableBlock = React.forwardRef((props, ref) => {
+    >
+      {children ?? (
+        <>
+          {expandIcon && <ExpandableBlock.ExpandIcon />}
```
typo?

```suggestion
          {expandIcon ?? <ExpandableBlock.ExpandIcon />}
```

LostABike replied:
`??` doesn't work I think because there might not be any value but just a prop pass in like this 
![image](https://github.com/iTwin/iTwinUI/assets/22462084/eff2f923-365d-4ab1-9769-1a4f23672246)

You can see that it doesn't show up here https://itwin.github.io/iTwinUI/1354/react/?path=/story/core-expandableblock--basic

mayank99 replied:
But why would you want to pass it without a value? it's supposed to be a reactnode, not a boolean.
![](https://github.com/iTwin/iTwinUI/assets/9084735/9437e147-6d14-4524-8f1a-932afe7dfb25)

On that note, I'm unsure if we even need this prop. Is there a case where you're trying to render the header without ExpandIcon or with a custom one?

LostABike replied:
Yea I think we talked about this
![image](https://github.com/iTwin/iTwinUI/assets/22462084/21b92c54-4213-4b2b-871a-17c35583cf82)

So I tried to make the icon customizable, but that makes the code too extra so I tried to add a convenience option as well

LostABike replied:
Yea I think we talked about this
![image](https://github.com/iTwin/iTwinUI/assets/22462084/21b92c54-4213-4b2b-871a-17c35583cf82)

So I tried to make the icon customizable, but that makes the code too extra so I tried to add a convenience option as well

mayank99 replied:
Oh I see. So you can keep the `expandIcon: React.ReactNode` if you want. But _imo_, it is also ok to leave it - it will still be possible to customize it using children. It is not that often (if at all) that you want to change the icon, so extra code is ok. (In my original comment, what I really meant by "customizable" is adding a custom css classname, not changing the icon itself.)

mayank99 replied:
If you're keeping `expandIcon: React.ReactNode`, then all stories and examples need to be changed to remove it, because it should not be passed without a value.

LostABike replied:
but isn't that how it is used? to be passed in so that default icon will display? Or should we change it to be `boolean` instead?

mayank99 replied:
no, basic usage should be like this:

```jsx
<ExpandableBlock>
  <ExpandableBlock.Header label='Expandable block' />
  <ExpandableBlock.Content>Content in block!</ExpandableBlock.Content>
</ExpandableBlock>
```

LostABike replied:
I'm still confused, then `expandIcon` is not used at all?

mayank99 replied:
It should only be used if the icon needs to be changed. That's what I've been trying to say in this whole thread. See above: https://github.com/iTwin/iTwinUI/pull/1354#discussion_r1247110364

Custom usage for a ReactNode would be like this:
```jsx
<ExpandableBlock>
  <ExpandableBlock.Header label='Expandable block' expandIcon={'➡️'} />
  <ExpandableBlock.Content>Content in block!</ExpandableBlock.Content>
</ExpandableBlock>
```

And by default, it should use the caret icon (this is achieved using `??`). It doesn't make sense to ask users to always pass `expandIcon` for the default caret icon; we should include it by default because that's what is desired in 99% of cases.

LostABike replied:
ok yea that makes sense. I have updated all stories and website example and tested working as expected

LostABike commented:
> > > > Did you mean manually test NVDA on components with `aria-`?
> > > 
> > > 
> > > NVDA screen reader and keyboard navigation
> > 
> > 
> > Tested and verified everything working!
> 
> Can you post a screen recording testing all states (expanded, collapsed, disabled)? See related thread: [#1321 (comment)](https://github.com/iTwin/iTwinUI/pull/1321#discussion_r1213522621)
> 
> > ```
> > * Changesets: currently having trouble with this due to the following error:
> >   ![image](https://user-images.githubusercontent.com/22462084/248515639-2c7befeb-437a-44bb-a5d2-8209a61e80aa.png)
> > ```
> 
> Are you still facing this issue? Should have been fixed after `main` was merged into `dev` on Monday.

![expandable block test](https://github.com/iTwin/iTwinUI/assets/22462084/64db84ea-4857-4aef-be40-8d7b4f8c92f1)

mayank99 commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -63,45 +63,79 @@ type ExpandableBlockProps = {
   const [expanded, setExpanded] = React.useState(isExpanded);
   React.useEffect(() => {
     setExpanded(isExpanded);
   }, [isExpanded]);
```
Unresolving this comment because I think you've reverted part of this change.

The logic here is that `isExpanded` (prop) should always be preferred over the internal state. If the user wants default uncontrolled behavior, then `isExpanded` prop should not be passed.

LostABike replied:
hmm not sure I understand correctly. do you mean that the user controls the states themselves via `isExpanded` prop and their own `onClick`? 
the reason I reverted this was because the expanding and collapsing logic relies on the variable controlled by `useState` (which is `expandedState` in the suggested update). Leaving them as `expanded` won't change inner logic when clicking on header

mayank99 replied:
> the user controls the states themselves via `isExpanded` prop and their own `onClick`?

Yes but only if they want to opt into controlled mode. By default it will work correctly as uncontrolled.

> Leaving them as expanded **won't** change inner logic when clicking on header

It **will** change if `isExpanded` prop is not passed.


You can test this snippet in vite playground (note there is no `isExpanded` prop passed):

```jsx
<ExpandableBlock>
  <ExpandableBlock.Header label='Expandable block' />
  <ExpandableBlock.Content>Content in block!</ExpandableBlock.Content>
</ExpandableBlock>
```

---

I think in storybook you are explicitly setting value of `isExpanded` in args. That needs to be removed if you want correct uncontrolled behavior.

LostABike replied:
Ok yea those controls made me think the logic was off. I reverted this change back. Also, removed `isExpanded` toggle logic for Accordion Story since that seems to not be needed anymore?

mayank99 commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -63,100 +64,259 @@ type ExpandableBlockProps = {
+
+  const icon = children ?? (status && StatusIconMap[status]());
+  return (
+    <Icon className={className} fill={status} ref={forwardedRef} {...rest}>
```
if there is no additional class here, then `className` doesn't need to be destructed, it will be used automatically through `...rest` props.

also, `iui-status-icon` can be removed from scss/html code

LostABike replied:
thanks, updated and removed

mayank99 commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -63,100 +64,259 @@ type ExpandableBlockProps = {
+          {
+            'iui-expanded': expanded,
+            'iui-small': size === 'small',
+            'iui-borderless': styleType === 'borderless',
```
`iui-expanded`, `iui-small` and `iui-borderless` could be data attributes

https://github.com/iTwin/iTwinUI/pull/1354/files/924a3064d78e60073950372b6138023367570f5d..001d046d306da6846b67f7765cf7573715ce348f#r1246683447

LostABike replied:
these class names have been updated :d

mayank99 replied:
i was thinking converting them to data attributes would be better than class names, like so:

```html
<div class='iui-expandable-block' data-iui-size='small' data-iui-expanded='false' data-iui-disabled='true'>
```

you can see an example in `button.scss`+`Button.tsx`

LostABike replied:
ohhh o_o

LostABike replied:
Done! had some trouble with fixing tests (that also came from other changes not just this in particular), but @adamhock helped me :D

mayank99 commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -135,7 +135,7 @@ const ExpandableBlockHeader = React.forwardRef((props, forwardedRef) => {
       type='button'
       className={cx('iui-expandable-header', className)}
-      aria-expanded={isExpanded}
+      aria-expanded={isExpanded ?? !disabled}
```
why?

```suggestion
      aria-expanded={isExpanded}
```

LostABike replied:
I was trying to pass this unit test 

```
it('should respect disabled prop', () => {
  const onToggleMock = jest.fn();
  const { getByRole } = render(
    <ExpandableBlock onToggle={onToggleMock} disabled>
      <ExpandableBlock.Header label='test title' />
      <ExpandableBlock.Content>test content</ExpandableBlock.Content>
    </ExpandableBlock>,
  );
  const header = getByRole('button');
  expect(header).toHaveAttribute('aria-disabled', 'true');
  fireEvent.click(header);
  expect(onToggleMock).not.toHaveBeenCalled();
  expect(header).toHaveAttribute('aria-expanded', 'false');
});

```

Since passing `disabled` only adds `aria-disabled` attribute and not `aria-expanded`, I try to explicitly add it this way when `isExpanded` is not present

mayank99 replied:
discussed offline and fixed in [`65ba13b`](https://github.com/iTwin/iTwinUI/pull/1354/commits/65ba13b0dc9d475ec2a034b264f83186e7aa0362)

mayank99 commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -63,100 +64,254 @@ type ExpandableBlockProps = {
+  const { isExpanded } = useSafeContext(ExpandableBlockContext);
+
+  return (
+    <WithCSSTransition in={isExpanded}>
```
Tangential to API changes: looks like animation is not working, because the `WithCSSTransition` component doesn't take #1302 into consideration. Might be time to use [pure CSS transitions](https://css-tricks.com/css-grid-can-do-auto-height-transitions/).

It can be done in a separate PR but you might want to start working on that PR *now* in case it requires any API/dom structure changes. You can create a chained PR by creating a branch on top of `expandable-block-composition` and using it as the base branch for your PR.

mayank99 commented on packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx:
```diff
@@ -166,15 +166,16 @@ ExpandableBlockHeader.displayName = 'ExpandableBlock.Header';
-    <Box
+    <Icon
       className={cx('iui-expandable-block-icon', className)}
+      size='medium'
```
i think this size prop can be removed

LostABike replied:
removed

mayank99 commented on packages/itwinui-css/src/expandable-block/expandable-block.scss:
```diff
@@ -125,8 +125,6 @@
 
 /// @arg $type Can be one of: expander, status
 @mixin iui-expandable-block-header-icons($type) {
-  @include mixins.iui-icon-style('m');
```
below, instead of setting `inline-size`, `block-size` and `fill`, you can set these css variables.

```scss
--iui-svg-size: var(--_iui-expandable-block-expander-icon-size);
--iui-svg-fill: var(--_iui-expandable-block-expander-icon-fill);
```

LostABike replied:
thanks! this works great for adjusting the icon size based on attribute. I was going to go `:where` route..

mayank99 commented on packages/itwinui-css/backstop/tests/expandable-block.html:
```diff
@@ -341,9 +372,9 @@ <h2>Borderless</h2>
+      document.querySelectorAll('.iui-expandable-header:not([aria-disabled="true"])').forEach((el) => {
         el.addEventListener('click', (event) => {
-          event.currentTarget.parentElement.classList.toggle('iui-expanded');
+          event.currentTarget.parentElement.classList.toggle('[data-iui-expanded="true"]');
```
this code doesnt work. you need to set the attribute, not class.

something like this might work (untested):
```suggestion
        const prevExpanded = event.currentTarget.parentElement.dataset.iuiExpanded;
        event.currentTarget.parentElement.dataset.iuiExpanded = !prevExpanded;
```

another option is to use `setAttribute`

mayank99 commented on packages/itwinui-css/src/expandable-block/expandable-block.scss:
```diff
@@ -148,77 +133,65 @@
-  @if $type == 'sublabel' {
-    font-size: var(--iui-font-size-0);
-    color: var(--_iui-expandable-block-header-caption-color, var(--iui-color-text-muted));
+.iui-expandable-block-caption {
```
some of these styles are the same as above and can be combined, e.g.:

```scss
:is(.iui-expandable-block-title, .iui-expandable-block-caption) {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  transition: color var(--iui-duration-1) ease;
  @media (forced-colors: active) {
    transition: none;
  }
}

.iui-expandable-block-caption {
  font-size: var(--iui-font-size-0);
  color: var(--_iui-expandable-block-header-caption-color, var(--iui-color-text-muted));
}
```

mayank99 replied:
still waiting for this

LostABike replied:
updated!

LostABike commented on apps/storybook/cypress-visual-screenshots/baseline/ExpandableBlock.test.ts-Accordion (Closed).png:
manually adding `line-height: inherit` to header does bring header height back to be the same as dev's but content is still shorter

LostABike replied:
so I updated `font-family` to `font` and it does help keeping the header same height, not content.

according to the computed layout it tells me that `line-height` is where the discrepancy is:
dev:
![image](https://github.com/iTwin/iTwinUI/assets/22462084/69e3c4ca-5737-45e6-940e-788e54fc2b6e)
local:
![image](https://github.com/iTwin/iTwinUI/assets/22462084/340dfca8-d7b6-42af-a7f8-82332e15276c)

everything else remains consistent.
so I added `line-height` to content and that seems to adjust the height to be same as on dev o_O

LostABike replied:
that is odd. without that `line-height` content is shorter than `dev` for me, but let me try that again

mayank99 replied:
this is the url for `dev` (v3): https://itwin.github.io/iTwinUI/dev/react/?path=/story/core-expandableblock--basic

this is `main` (v2): https://itwin.github.io/iTwinUI/dev/react/?path=/story/core-expandableblock--basic

LostABike replied:
wow. thanks for those! I've been just trying to re-run the tests and hope something click

LostABike replied:
ya new block's content is slightly shorter than current test images. approve?

mayank99 replied:
weird that it's only in css, but yeah approve.

before approving, can you make sure all headers in the html file use `<button>`? i'm seeing some of them are still using `<div>`

LostABike replied:
updated. and css tests are still passing at least locally. lets see

mayank99 reviewed (approved):
LGTM
